api/pkg/artifact/artifact.go

```diff
@@ -69,6 +80,37 @@ func (gac *GcsArtifactClient) ParseURL(gsURL string) (*URL, error) {
 	}, nil
 }
 
+type Service interface {
+	GetType() string
+	GetURLScheme() string
+	ParseURL(gsURL string) (*URL, error)
+	ReadArtifact(ctx context.Context, url string) ([]byte, error)
+	WriteArtifact(ctx context.Context, url string, content []byte) error
+	DeleteArtifact(ctx context.Context, url string) error
+}
+
+type GcsArtifactClient struct {
+	URLScheme
+	Type string
+	API  *storage.Client
+}
+
+func NewGcsArtifactClient() (Service, error) {
+	api, err := storage.NewClient(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("%s,failed initializing gcs for the artifact client", err.Error())
```

Comment:
nit: should it be `"errorSubject: errorDetail"`? e.g `fmt.Errorf("failed initializing gcs for the artifact client with error: %s,", err.Error())

Reply:
Oh yes good point, thanks for picking that up. I can't remember how I came up with such a strange error message 😅